sql/src/main/java/io/crate/metadata/sys/SysJobsTableInfo.java

```diff
@@ -35,58 +34,43 @@
 import io.crate.metadata.expressions.RowCollectExpressionFactory;
 import io.crate.metadata.table.ColumnRegistrar;
 import io.crate.metadata.table.StaticTableInfo;
-import io.crate.types.DataTypes;
 import io.crate.types.ObjectType;
 import org.elasticsearch.cluster.ClusterState;
 import org.elasticsearch.cluster.node.DiscoveryNode;
 
+import java.util.Map;
 import java.util.function.Supplier;
 
+import static io.crate.types.DataTypes.STRING;
+import static io.crate.types.DataTypes.TIMESTAMPZ;
+
 public class SysJobsTableInfo extends StaticTableInfo {
 
     public static final RelationName IDENT = new RelationName(SysSchemaInfo.NAME, "jobs");
-    private static final ImmutableList<ColumnIdent> PRIMARY_KEY = ImmutableList.of(Columns.ID);
 
-    public static class Columns {
-        public static final ColumnIdent ID = new ColumnIdent("id");
-        static final ColumnIdent USERNAME = new ColumnIdent("username");
-        static final ColumnIdent STMT = new ColumnIdent("stmt");
-        public static final ColumnIdent STARTED = new ColumnIdent("started");
-        static final ColumnIdent NODE = new ColumnIdent("node");
-        static final ColumnIdent NODE_ID = new ColumnIdent("node", "id");
-        static final ColumnIdent NODE_NAME = new ColumnIdent("node", "name");
+    static Map<ColumnIdent, RowCollectExpressionFactory<JobContext>> expressions(Supplier<DiscoveryNode> localNode) {
+        return columnRegistrar(localNode).expressions();
```

Comment:
this builds the `ColumnRegistrar` on each query and it was already instantiated at the ctor. we should reuse the one of the ctor.

Reply:
Sure, we can do this optimisation. Although the expression were initially also build on every single call from scratch, so I guess it won't add any overhead to the previous implementation.

Reply:
The expressions resulted in 1 map while the using the ColumnRegistrar will create 1 instance of it and internally 3 maps and 2 other instances. So it will add quite some allocation overhead. Although I think reusing it is quite simple to implement and adds no complexity.

Reply:
>  In some cases there are parameters passed to the expression method and then we need to rebuild it anyway each time.

Hm don't get that, the new cases I saw are requiring a  `localNode` supplier which is fine to reuse from the ctor. Maybe you can point me to other problematic cases?

Reply:
ok, i just did a little microbenchmark, it is a lot faster to reuse the ColumnRegistrar. Thanks for the feedback.

Reply:
> > In some cases there are parameters passed to the expression method and then we need to rebuild it anyway each time.
> 
> The new cases I saw are requiring a `localNode` supplier which is fine to reuse from the ctor. Maybe you can point me to other problematic cases?

I reviewed the cases, it should be possible to implement it the way you suggested.

Reply:
It is currently not possible to reuse the ColumnRegistrar from the ctor, since 
expressions is a static function. However, to do so expressions could be part 
of a StaticTable, and StaticTable would need to be parameterized according to 
the return type of the expression. The SchemaInfo and TableDefinition could then
be merged together since they are more less doing the same anyway - Holding
a reference to a table based on the name. However, I believe that the current
implementation adds very little overhead to the original implementation.
As far as I understood, the expression method is not called on each query, 
but rather initialized once in TableDefinition on startup and kept in memory.

Reply:
Yes you're right, it's just called once. Anyway I think the current implementation is a bit hacky as we're forced now to pass in arguments at the ctor of the table infos while they are not used at all (and also that we create the column registrar twice).
Ok lets keep it like that and try to move on to better solutions out of this PR.

Reply:
> and StaticTable would need to be parameterized according to
the return type of the expression
To fix already existing Generics issues, the StaticTable should be parameterized according to the given row type now already.

Reply:
> The SchemaInfo and TableDefinition could then
be merged together since they are more less doing the same anyway - holding
a reference to a table based on the name.

I guess you mean `(Static)TableInfo`.
The current definition holds expressions for collecting rows of a static table which is another scope than holding the reference definitions (column definitions) of a table. (this is the table metadata).
I think we should not mix this separation but more going into the direction of treating static tables similar to dynamic once. One solution may be to map concrete static table iterables to `Row`s and plan inputs for all queried/required columns.

Reply:
> The current definition holds expressions for collecting rows of a static table which is another scope than holding the reference definitions (column definitions) of a table. (this is the table metadata).
> I think we should not mix this separation but more going into the direction of treating static tables similar to dynamic once. One solution may be to map concrete static table iterables to `Row`s and plan inputs for all queried/required columns.

Sounds good to me, lets do this in another pr.